packages/veritone-widgets/src/widgets/FolderTree/index.js

```diff
@@ -245,19 +254,21 @@ function FolderTreeWrapper({
 
   //modify folder
   const handleSubmitModify = (selectedFolder = {}, folderName) => {
+    handleCloseModify();
     const currentFolderId = currentFolderForAction.id;
     const newParentId = Object.keys(selectedFolder)[0];
     if (openEdit) {
       editFolder(currentFolderId, folderName, true, false);
     } else {
       editFolder(currentFolderId, folderName, false, true, newParentId);
     }
-    handleCloseModify();
   };
 
   const handleCloseModify = () => {
     setOpenModify(false);
-    setOpenEdit(false);
+    setTimeout(() => {
+      setOpenEdit(false);
+    }, 500)
```

Comment:
Why do we have a delay here?

Reply:
I want the dialog is disappeared completedly before we change the data.